Include/Acidanthera/Library/OcBootManagementLib.h

```diff
@@ -52,12 +52,15 @@ typedef struct OC_HOTKEY_CONTEXT_ OC_HOTKEY_CONTEXT;
 #define OC_MENU_DISK_IMAGE           L" (dmg)"
 #define OC_MENU_SHUTDOWN             L"Shutting Down"
 #define OC_MENU_RESTART              L"Restarting"
+#define OC_MENU_SIP_IS_DISABLED      L"SIP: Disabled"
+#define OC_MENU_SIP_IS_ENABLED       L"SIP: Enabled"
```

Comment:
It is very confusing when these are labelled "Enable SIP" and "Disable SIP" - or at least I found it so.

Reply:
Is okay to resolve this one and go with these labels? Esp. since you've already pinged Andrey and I guess he's currently producing audio corresponding to it this way - but just to double-check it's okay to go with this? Thx.

Reply:
This looks fine to me, but maybe Toggle SIP ON/ Toggle SIP OFF is more intuitive to be honest. Because a clickable entry is an action in the first place.

Reply:
I know it's an action, but the thing is, when the label says ON, or Enable, that is the first thing you see (and that would mean it is off!!). Basically a label which describes what effect the action will have necessarily describes the opposite of the current state - which is confusing when you actually use it. I do appreciate that it negates the logic of describing what the action will be, but I have tried both, and this way round is much easier in use.

Reply:
Well, maybe @mhaeuser can say a word or two, I think I am ok with the suggested code.

Reply:
"Toggle SIP ON/ Toggle SIP OFF" does not make it perfectly clear whether ON/OFF is the state or action in my opinion, I don't think "toggle on/off" is common language (rather "turn on/off"). I don't see the issue with "Enable/Disable SIP".

Reply:
As I said, I actually used "Enable/Disable SIP" and I found it confusing to use. When it says "Enable SIP" it is telling you (what you probably want to know) that SIP is currently disabled, and vice versa. (Like I said, if you name it the action it will take, then you necessarily name it the opposite of the current state.) I am not a fan of changing to "ON/OFF" in the wording either.

I would be happy, lets say, to choose between "Enable SIP"/"Disable SIP", and "SIP: Disabled"/"SIP: Enabled". But in actual day-to-day use, I found that the one which shows the _state_ ("SIP: Disabled"/"SIP: Enabled") is much preferable to the one which shows what the next action will be ("Enable SIP"/"Disable SIP") (since that necessarily has wording which reflects the _opposite_ of what the current state is).

Reply:
It obviously usable either way - it's just nicer (I believe, having used both) with the way which shows the state.

Reply:
(FWIW - It gets even more confusing when you add in checking how it's working, as SIP is Enabled when it is zero and disabled when it is non-zero, so there is a profusion of double-negatives around here!)

Reply:
@MikeBeaton Well, this is a debugging utility, you should turn SIP off only shortly to complete your testing. OC is not really a dashboard where you monitor your system's state, this is an action like any other to me.

Reply:
(You need SIP for some other things actually, like even setting flavours on macOS boot directories - though (fortunately) that's not really needed much with the auto-detection of macOS version.)

So I mean, as ever, if I'm implementing something at least partly because I'd like to use it, then I'd *rather* implement the version I'd actually like to use. As I said, the naming I've switched to (at least for now, after using the initially suggested ones initially) do make it more usable, to me.

But ofc I'm not going to not do it, if there's a strong feeling from you and Vit that I should not do the version I'd really prefer...! I gather there's quite a strong feeling from you, and so far a somewhat neutral feeling either way from @vit9696 . Hmmmmm....

Reply:
An alternative is to maybe make some text feedback about the state, after selecting the action. But - again - I think it is quicker and simpler to use without that, i.e. it is basically pretty much quick, simple and intuitive to use as it is now...

Reply:
Maybe Toggle SIP (Enabled) / Toggle SIP (Disabled)?

Reply:
Last comment - but - obviously with most boot entries, there's a specific label which says what it will do - which is always what it will do. (For this, this label would be "Toggle SIP".) But this one specifically needs to (or rather, does - as a convenience) let the user know what the current state is, in order to be sure what it is going to do. So it either implicitly tells the user the current state ("Enabled SIP" tells the user SIP is disabled, "Disable SIP" tells the user SIP is enabled), or it does it more explicitly. So there is some motivation for treating this one differently - in that it's already different in this way, more or less, whichever of the two options under discussion we use... hope that verbose-ness also makes some kind of sense!

Reply:
Okay - thank you wise leader! Will do that.